src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2461,11 +2461,19 @@ class StubGenerator: public StubCodeGenerator {
   }
 
   // code for comparing 8 characters of strings with Latin1 and Utf16 encoding
-  void compare_string_8_x_LU(Register tmpL, Register tmpU, Register strL, Register strU, Label& DIFF) {
+  void compare_string_8_x_LU(Register tmpL, Register tmpU,
+                             Register strL, Register strU, Label& DIFF) {
     const Register tmp = x30, tmpLval = x12;
+
+    int base_offset = arrayOopDesc::base_offset_in_bytes(T_CHAR);
+
+    assert((base_offset % (UseCompactObjectHeaders ? 4 :
+                           (UseCompressedClassPointers ? 8 : 4))) == 0, "Must be");
+
+    // strL is 8-byte aligned
     __ ld(tmpLval, Address(strL));
```

Comment:
This should also be a `load_long_misaligned`?

Reply:
And the below one in this function?

Reply:
Good catch! I think I missed the second one. Fixed.
For the first one, I think we are OK as `strL` is ensured to be 8-byte aligned by the caller [1].

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/stubGenerator_riscv.cpp#L2526